src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -158,12 +161,12 @@ protected void FinishDocumentSetup(Document doc, Lucene.Net.Search.ScoreDoc scor
 
                 Dictionary<string, FieldsToFetch.FieldToFetch> fields = null;
 
-                if (SearchEngineType == SearchEngineType.Lucene)
+               
+                if (FieldsToFetch.ExtractAllFromIndex)
                 {
-
-                    if (FieldsToFetch.ExtractAllFromIndex)
+                    if (retrieverInput is not {LuceneDocument: null, State: null})
```

Comment:
Same questions about the condition like above. Maybe we should have some method like `IsLuceneDocument()`?

Reply:
Good idea! Done